src/d3d12/visualizer.rs

```diff
@@ -271,8 +278,17 @@ impl AllocatorVisualizer {
         self.focus = None;
     }
 
-    pub fn render(&mut self, allocator: &Allocator, ui: &imgui::Ui) {
-        self.render_main_window(ui, allocator);
-        self.render_memory_block_windows(ui, allocator);
+    pub fn render(&mut self, allocator: &Allocator, ui: &imgui::Ui, opened: Option<&mut bool>) {
+        let draw = if let Some(o) = opened.as_ref() {
+            **o
+        } else {
+            true
+        };
+
+        self.render_main_window(ui, opened, allocator);
+
+        if draw {
+            self.render_memory_block_windows(ui, allocator);
+        }
```

Comment:
Then, given the comment above that we shouldn't call `render_main_window` if it's `Some(false)` (ie. the user/caller _explicitly_ - by passing `Some` - asks for this window to _not_ be shown) you can massively simplify this with:

```suggestion
        if opened != Some(&mut false) {
            self.render_main_window(ui, opened, allocator);
            self.render_memory_block_windows(ui, allocator);
        }
```

If the caller doesn't care about showing or hiding the window and passes `None` this code will always run (and as discussed offline, not calling `.opened()` causes the `x` button to remain hidden).

By the way, don't forget to add some doc-comments explaining this use :)